archunit/src/test/java/com/tngtech/archunit/library/dependencies/SliceRulePerformanceTest.java

```diff
@@ -27,31 +34,12 @@ public void searching_for_cycles_should_terminate_reasonably_fast_for_complete_g
         SliceRule cycleFree = slices()
                 .matching(completeGraphPackageRoot + ".(*)")
                 .should().beFreeOfCycles();
-
-        String violations = cycleFree.evaluate(classesFormingCompleteDependencyGraph).getFailureReport().toString();
-        int numberOfDetectedCycles = countOccurrences(violations, "Cycle detected: ");
         int expectedNumberOfCycles = getNumberOfCyclesInCompleteGraph(numberOfClassesFormingCompleteGraph);
-        assertThat(numberOfDetectedCycles).as("number of cycles detected").isEqualTo(expectedNumberOfCycles);
-    }
+        ArchConfiguration.get().setProperty(MAX_NUMBER_OF_CYCLES_TO_DETECT_PROPERTY_NAME, String.valueOf(2 * expectedNumberOfCycles));
 
-    private int countOccurrences(String text, String toFind) {
-        return Splitter.on(toFind).splitToList(text).size() - 1;
-    }
+        String violations = cycleFree.evaluate(classesFormingCompleteDependencyGraph).getFailureReport().toString();
 
-    /**
-     * To calculate the number of expected cycles check the following (we denote binomial coefficients "n over k" as (n k) ):
-     * For any k-cycle (cycle of k nodes within a graph of n nodes) we can denote
-     * n1 -> n2 -> n3 -> ... -> n[k-1] -> n[k] -> n1
-     * Fix n1 and we have (k-1)! possibilities to order the other k-1 nodes.
-     * Now we have (n k) possibilities to choose k nodes to form a k-cycle and for each choice (k-1)! permutations.
-     * Thus we retrieve the sum of all cycles as
-     * SUM(k=2, n) (n k) * (k-1)! = SUM(k=2, n) n! / ((n-k)! * k)
-     */
-    private int getNumberOfCyclesInCompleteGraph(int numberOfNodes) {
-        int numberOfCycles = 0;
-        for (int k = 2; k <= numberOfNodes; k++) {
-            numberOfCycles += factorial(numberOfNodes) / (factorial(numberOfNodes - k) * k);
-        }
-        return numberOfCycles;
+        int numberOfDetectedCycles = countCyclesInMessage(violations);
+        assertThat(numberOfDetectedCycles).as("number of cycles detected").isEqualTo(expectedNumberOfCycles);
```

Comment:
Should we really assert that in this test category? I would expect the assertion on the number of cycles in the unit test `GraphTest` if at all.

Reply:
As mentioned, can argue about that. I mean the point is that the test terminates in time, I know. But a test without assertion seems confusing, too, somehow.
We could write a comment instead, like "if we reach this point in reasonable time, we consider the performance test passed"? :wink: 
Or could simply assert `violations != null`, but that would also be weird...

Reply:
I see the point of the test without assertion. We can leave that here (as the test hopefully never breaks because of deviation of numbers here).